### ISSUE TITLE
Error during WebSocket handshake: Unexpected response code: 400

### ISSUE THREAD

**Issue by centrual** [scalecube/socketio#39]
Hello,

I trying to make Socket.IO server with Java. I made sample app but when i try to connect `WebSocket connection to 'ws://localhost:3132/socket.io/?EIO=3&transport=websocket' failed: Error during WebSocket handshake: Unexpected response code: 400` message coming back.

I using http://amritb.github.io/socketio-client-tool/ for test.

What should i do?

Thank you so much.
Have a great day!

**Comment by centrual** [comment]
Why we not upgrade server's socket.io client support to latest?

**Comment by antonkharenko** [comment]
Basically the need was to have a high performance feature-less websocket transport with possible fallbacks (in case when ws is not supported by browser or blocked by proxy/firewall). Historically it was implemented when 0.9.x version was the latest and it was so for pretty long time. Later on once 1.X versions of socket.io protocol was introduced it was moved in the direction of supporting more higher level features (e.g. session management and so) which I think should not be done on transport layer and also it changed the logic from fallback from websocket to other transport to upgrade where it originally start from some lower level transport and then try to upgrade to websocket. Basically all this wasn't aligned with the original motivation to make Socket.IO server and also due to significant changes to protocol it require quite a lot of rewrite so we decided to stay on 0.9.x version of Socket.IO protocol.
